### DISCUSSION TITLE
Ability to keep remaining team members as reviewers after one team member starts a review

### DISCUSSION THREAD

**Discussion by nieuwsma** [community/community#5289] [Pull Requests]
Presently, if I add a team as a reviewer on a pull request, the first person from that team to start a review does so "on behalf of" the team, and the remaining team members no longer see that their review is requested on that pull request.

I can see that this may be useful in some cases, however in our company's case, we require at least one review from a team member, but encourage other team members to review also. Therefore, we would like a setting for all team members to remain on the PR as reviewers. I could see this being a repo-level or org-level setting.

Our workaround has been to list each individual in the CODEOWNERS files, however this is suboptimal because when a team changes, we will need to update the list of team members in multiple CODEOWNERS files, instead of in the central configuration for that team.

TAKEN from: https://github.com/isaacs/github/issues/1211 

a LOT of people would really like this 'feature' fixed. This makes management of PRs email driven rather than github UI driven.

**Comment by alexander-anderson-salesforce** [comment]
This is on the first page of all feature requests when sorting by TOP.  Doesn't seem like it is getting surfaced/noticed.  My team would also love to have this feature.

**Reply by simplisamtwining** [reply]
This is also a major annoyance when trying to find PRs that you've been tagged in on our team. Insight into PRs that have a team tagged that someone has touched is aggravatingly low.

**Comment by aitkin-dn** [comment]
So there is no solution to this issue currently? 
Our team was so happy that we don't need to add everyone manually when we discovered this group option, but now we understand we can't really use it as we did the previous review requests so that everyone will be informed even if someone reviewed :\

**Reply by Snailedlt** [reply]
It is possible to enable the option to add every member of the group when the group is selected as reviewer.

Depending on your usecase this could fit you

**Reply by Snailedlt** [reply]
Ref: https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team#configuring-auto-assignment

**Reply by priscillasantana** [reply]
It's limited to 3 team members

**Reply by Snailedlt** [reply]
Bummer :(
Why would they put in place such a limit? Seems like unessesary restrictions to me

**Reply by ay-azara** [reply]
They bumped it up to 7

**Reply by Snailedlt** [reply]
Wonder why they have a limit at all

**Reply by shyamsundar** [reply]
> They bumped it up to 7

Can you point me to the docs where it says 7? We see 3 on our server, wondering what version it is.

**Reply by ay-azara** [reply]
@shyamsundar It doesn't specifically say in the docs but the Github website, for me, shows me up to 7 under Team->Settings->Code Review. Since you said server, I assume you're self-hosting, so maybe it is not up to date with the latest version available? That's just speculation on my part. SnailedIt has a screenshot further down of what it should look like to you.

**Reply by Snailedlt** [reply]
For future reference: The screenshot can be found here: https://github.com/orgs/community/discussions/5289#discussioncomment-7777105

**Reply by shyamsundar** [reply]
Thanks @Snailedlt. It looks like our server is not on the latest release. I only see 3 in the dropdown and I also don't see the 'Scheduled reminders' option that's in the screenshot.

**Reply by shyamsundar** [reply]
@Snailedlt, @ay-azara  any idea from which version this setting allows `7` instead of `3`?

**Reply by Snailedlt** [reply]
Nope, no idea :/

**Comment by samibeyoglu** [comment]
This is really annoyance! Every time I comment on a PR,  my team gets removed from the reviewers and I have to re-add it.

**Comment by jeffhelps** [comment]
This works fine for single approver repos, but clearly doesn't when you have 2 (or more) approvers needed, would like to see this get fixed

**Comment by simplisamtwining** [comment]
For anyone else vexxed by this: 

A coworker filled me in on the workaround by configuring auto-assignment. When your team is assigned it'll individually add members of your team.

https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team#configuring-auto-assignment

**Reply by michaelheyman** [reply]
This workaround only solves the problem stated in this thread for teams with up to 3 members, because there's a limit of 3 members that can be auto-assigned to review.

**Reply by himashan** [reply]
they bumped it to 7 now

**Comment by hubbcaps** [comment]
+1, this is desperately needed for repos that require more than 2 reviewers. PRs end up collecting dust due to not being visible to the larger team after someone hops on an initial review.

**Comment by Snailedlt** [comment]
+1.
I had no idea it even was like this. Imo, the default behavior should be that all reviewers stay reviewers unless you remove them as reviewers.

**Comment by andreas-hakansson-ingka** [comment]
+1, we also require at least 2 reviewers and created a team to help assign the correct people. Our expected behavior wasn't that everyone else would be removed as soon as a single developer performed a review. As the bare minimum, make this behavior a branch protection setting so we can choose

**Comment by drewnichols** [comment]
+1 I would also like to be able to add my entire team to the PR more easily. Having the team removed after 1-3 people review the PR is not helpful.

**Comment by melvincwng** [comment]
+1 agree on this feature request. Hope to see it being implemented in the near future 👍🏻

**Comment by max-pi** [comment]
I think default behavior should be that team stays assigned as reviewers, as this more closely matches behavior when you specify specific reviewers. Or ideally this should be a toggle setting for the team

**Comment by ClementHoang-at** [comment]
:+1: Would like this so that the flow can less email driven.

**Comment by kemallette** [comment]
This was a surprising discovery for our team. Please address.

**Comment by jzisser9** [comment]
Our team was so excited when we discovered this feature, but the fact it de-selects the group when one person responds to the PR is a huge deal-breaker for us. Please address this - this is an amazing feature, especially for larger teams, hampered by one small aspect of its design.

**Comment by pawelantosik** [comment]
+1

**Comment by StevenHarlow** [comment]
+1 For our team

**Comment by ravindrasinghshah** [comment]
+1

**Comment by vparracortes01** [comment]
+1

**Comment by stiliyan-angelov-mercell** [comment]
Same issue with our team.

**Comment by sebthibert** [comment]
My life feels empty without this feature.

**Comment by wallentx** [comment]
My team would also appreciate this

**Comment by maxnorth** [comment]
Same, was very surprised to see the team auto-removed when one person approved. Depending on the complexity of the PR we generally expect at least a couple approvals. But we want everyone to see every PR and at least have a chance to take a look and review, so we don't want to restrict to only 3 auto-assigned members.

**Comment by mccormick-wooden** [comment]
+1

Alternatively, it would be great to have the option to be able to submit a review _not_ on behalf of any team, so that the team request isn't cleared.

**Reply by Snailedlt** [reply]
Additionally*
The original feature requested here is a must if you ask me, but this is good as a supplement though :)

**Comment by jeffcjlee** [comment]
+1

This is a deal breaker for my team.

**Comment by amytackett** [comment]
+1 the groups feature isn’t very useful without this

**Comment by gmiklich** [comment]
Others have mentioned the "auto-assign" functionality, which I understand doesn't fully resolve the issue (it doesn't for me either).  However, it does point to an alternative path for the Github team: change the max number of people to be assigned from 3 to the number of people in the team.  So if the team has 8 members, when we use the auto-assign feature, we should be able to select 1-8 instead of just 1-3.

**Reply by gmiklich** [reply]
As an update to this, I _think_ they increased the max number of people assigned from 3 to 7.  So it's a step in the right direction.  However, I'd love a maintainer to comment on why they don't just bind the drop down to the number of people in the team.  Why stop arbitrarily at 7?

**Comment by marina101** [comment]
+1 this would be very helpful for our team as we've sometimes had PR reviews delayed for a few days if folks forget to re-add the team for a second review

**Comment by TetraGG** [comment]
+1 it is annoying to re-add the group after the group review was automatically replaced by the reviewer.

**Comment by priscillasantana** [comment]
At least increase the number of team members that will be tagged. It would be already helpful!!

**Comment by philyawj** [comment]
+1

**Comment by jsdw** [comment]
+1!

Having the team not be removed automatically would match my expectations more closely.

I wrote some automation which looked for any PRs with my team assigned and added them to a project board. This obviously doesn't work so well if the team is unassigned as soon as anybody on it leaves a comment or review (we want at least 2 reviews)

**Comment by parker-vv** [comment]
+1

**Comment by tonglil** [comment]
Are there any Actions out there that can re-run when a review is done to re-assign the codeowners as reviews to the PR?

1. Someone reviews
2. GH removes the team
3. Action runs
4. Action adds the team again

**Reply by jcook-uptycs** [reply]
The GitHub token given to actions does not support being able to add a team review! 🤯 

So any action to support this would need to have a personal access token.

**Comment by Joker368** [comment]
bump; running into this issue as well. We use auto-assign to assign 3 members to the review but we also want to open it up for the rest of the team to see that there is a pending review and give them an opportunity to dive in and check it out if they have time.

**Comment by c0dev0id** [comment]
+1
we have groups of 20-30 people and reviews are optional. We want everyone of the group to be able to review. Please give us an option to keep the group in the reviewer list. (Github Enterprise in our case...)

**Comment by aslafy-z** [comment]
A workaround is to reference *every individual users* into the CODEOWNER file that will then configure automatic reviewers assignment. However, adding a group instead in this CODEOWNER file falls on the OP issue.

I don't like this solution but it does the thing in my little team of 5 as of today.

**Reply by RubenGamarrarodriguez-tomtom** [reply]
> A workaround is to reference _every individual users_ into the CODEOWNER file that will then configure automatic reviewers assignment. However, adding a group instead in this CODEOWNER file falls on the OP issue.

I think that is not idea, every time one member joins or leave the team it would require some automation to update, review and merge the codeowners in the 40+ repos that a team maintains and operate

**Reply by Snailedlt** [reply]
If someone makes that automation script, please do share so the rest can use and contribute to it :)

**Comment by aaron-oxford** [comment]
+1

Now that I've been added to the core review team group, I can't just give an individual approval any more. I'm signing off for the whole team.

**Comment by federico-caimi-rally** [comment]
+1

**Comment by cpinkerton-kpler** [comment]
The default behaviour here is too opinionated - it needs to change or allow some configuration of the behaviour

**Comment by jacktoussaint** [comment]
+1

**Comment by Joker368** [comment]
Not sure if something like this has been mentioned yet but for anyone who wants a work around for this you can try the following.

Example for a team with 9 engineers:
Create two children teams under your main github engineering team. (It doesn't have to be a child team but I found it's easier to do it this way for organization)

XYZCodeReviewTeam1 - Add 7 members (the max number for code review auto assignments per team)
XYZCodeReviewTeam2 - Add 2 members

Under the Code Review settings for each of these teams:
Enable auto assignment
Set "How many team members should be assigned to review?" to the max value (atm that is 7 -- meaning each child team can support up to 7 engineers and hence the reason for splitting the 9 engineers between a team of 7 and a team of 2).

As new engineers join your team you can add them to XYZCodeReviewTeam2 until you reach the limit of 7 before creating XYZCodeReviewTeam3 and so on and so forth.

The last caveat is that you must update the .github/CODEOWNERS file for every repo to include each of these teams as a code owner. Now every time you open a PR these two teams will be added automatically and then (because of "Enable auto assignment") the individual engineers will be auto assigned. Github currently supports up to 100 reviewers so this should work for most scenarios.

We just implemented this and it's working great. Feel free to reach out to me if you need assistance or want to discuss further.

**Reply by jorhett** [reply]
> Not sure if something like this has been mentioned yet but for anyone who wants a work around for this

_insert lots of manual team editing_

This is really not useful for any org large enough to have more than 7 members 👎   I can't think of a single org I work with (open source or not) that doesn't have SSO that defines the teams externally. No, nobody is going to spend time in an IDP hacking up teams to balance them correctly.

If we had that kind of time, we should migrate to Gitlab who allows us the freedom to define the rules as we desire.

**Comment by superwiss** [comment]
+1

Our ground rule for code reviews is that more than one person participates in the review activity.
However, because of this limitation, we got a problems.
I would like to see a feature or option added that would allow team members to remain reviewers even after review activity begins.
(Github Enterprise in our case...)

**Comment by NicoPampe** [comment]
+1 - not a very helpful comment other then checking if this has gotten any traction or solved. I've recently set up Team's in our company github flow for auto team assignment on PR's. It's very handy to get everyone on the PR's, but then everyone is lost once a single person reviews it, the Team addition is gone.

**Comment by rngadam** [comment]
We've got our pull requests for one repository set to at least three reviewers and a Github team (all) added,

as soon as one reviewer reviews, the team is cleared out even though the requirement hasn't been fulfilled

![image](https://github.com/community/community/assets/538542/de5c8afa-f8c0-4545-a726-04e358eb61b7)

**Reply by Snailedlt** [reply]
If you're 7 or less developers you can use a combination of these two settings under `https://github.com/orgs/your_org_name/teams/your_team_name/edit/review_assignment`
![image](https://github.com/community/community/assets/43886029/22f4d868-3d4a-4c16-96fb-096c29a6a59d)

**Comment by gabriellamichelle** [comment]
+1 The team scheduled reminder to slack is not that useful with this existing behaviour. Keeping the team would help lots of our workflows.

**Comment by rngadam** [comment]
> If you're 7 or less developers you can use a combination of these two settings under `https://github.com/orgs/your_org_name/teams/your_team_name/edit/review_assignment`

I can see how this can be helpful so thanks for ponting that out, but it's not exactly what we want.

* we have PR that has content relevant to the whole team
* we want to assign the PR to all
* as soon as we clear a set number of reviews (3) from anyone in the team, we can clear
* we still want others from the team to have a look beyond that

**Reply by Snailedlt** [reply]
I see. If you have multiple teams that can review the PR, but these rules only applies to one of them, then there's not currently any solution for this afaik. However if the goal is to just ensure that the PR is reviewed by 3 people before it gets merged, then you can use branch-restrictions instead:
![image](https://github.com/community/community/assets/43886029/1f6435a7-d82f-457c-ad5f-06904f83068b)

**Reply by rngadam** [reply]
Thank you!

**Comment by Edgar-Schmidt-Horsch** [comment]
+1 We need also every assigned member of the group to look up the pull request after one other member approved on behalf of the group. Otherwise the request will be lost.

**Comment by Wicaeed** [comment]
Why why why if I have a Team that is on a PR, and GItHub picks random users in that team to add directly to the PR, if one of the users that WASN'T picked and added to the PR comes and adds their review, WHY does the whole Team now get removed the PR?!?!

Useless, confusing behavior from GitHub...

**Comment by SergeyBarskiyTyler** [comment]
+1

**Comment by tamnaeuth** [comment]
+1

**Comment by ttt43ttt** [comment]
+1

**Comment by zeasey-apptio** [comment]
I've noticed this behavior recently and first thought it was a bug, why is the default to remove the team?

We've recently created a dashboard dedicated to view all open code reviews for our team, implemented with a search query.  
Example search:
```
org:MyOrg state:open type:pr team-review-requested:MyOrg/my-team-reviews
```
However, since "removing the team" is the default and can not be configured as disabled, it renders our dashboard pointless unless we continually add our team back on the review each time.

**Comment by robmcgee-isw** [comment]
+1 this makes the team reviewer approach useless to me

**Comment by PeterRydberg** [comment]
+1

It is very strange to _remove_ information from an existing PR, closed or not. If the team is requested for review, it should remain that way. I should be able to do `review-requested:@me` or at the very least `team-review-requested:ORG/TEAM` even if the PR is merged.

**Comment by kevindvnetapp** [comment]
+1.

For codebase where the files are reviewed by a large number of folks, automatically removing the team review is impacting us . The workaround listed above is limited by 7 people, which does not work for our team size.

**Comment by NetAppBlueDevil** [comment]
This behavior is crazy.    It says that the new intern in the team for the summer is equivalent to the senior engineer that has seen this code for 10+ years when commenting on a code review.    The requirement here is more than just wanting more people, it is specifically based on the mix of people in the team and experience level.     Fixing this requires simply a flag level option to wrap an if around the block of code that would remove the team from the PR.   This is not a "OMG we have to re-architect the entire thing" type of request.

**Comment by Thomas-Hagen-Shipt** [comment]
This is still an ongoing issue 5 years later. We are planning to stop using the team concept in our single-team CODEOWNERS and instead list out all of the individuals across all of our projects until support for this is added.

**Comment by jeorgmartin2** [comment]
Yeah, this limitation can definitely slow things down keeping all team members visible as reviewers would make collaboration much smoother and less dependent on manual workarounds like CODEOWNERS edits. Hope GitHub considers a flexible setting for this soon. Also, interesting discussion overall worth a read for dev workflows, just like exploring tools such as [Done55 Game Download](https://don55.net/) for different kinds of user experiences.

**Reply by davilima6** [reply]
this sounds like AI and should be deleted

**Comment by yoshi-jpc** [comment]
+1

**Comment by davidwarner234** [comment]
+1. Keeping the review request visible for all team members would encourage more reviews while still satisfying the required team approval. An optional org or repo setting would be a much better solution than listing every individual in CODEOWNERS.

**Comment by aws-mengchiy** [comment]
+1

**Comment by thatsanicehat** [comment]
Still asking for this 5 years later! The ability to set the number of reviewers would be ideal so that If I have a team of 15, I don't need to tag them individually but can still have that team's tag disappear after `n` members approve if I want.

**Comment by thescientist13** [comment]
Just coming in to say this a pretty big bummer after having just streamlined our project's to use teams for CODEOWNERS 😕

**Reply by robmcgee-isw** [reply]
Thanks for saving me the time. I thought that would workaround this problem.

**Comment by lucaspg-atscale** [comment]
We have a bot to notify on slack the open PRs for a team. There are repos that require 2 reviews. Every day, I was surprised by the fact that we didn't have PRs open for review (according to the bot), but the problem was the missing team :/

**Comment by crbunney** [comment]
I've worked around this by creating a simple workflow that triggers when a review is submitted and re-requests a review from the team, but it would be nicer if we didn't have to.

Here's the workflow if anyone wants to reuse it:
```yaml
on:
  pull_request_review:
    types:
      - submitted

permissions:
  pull-requests: write

jobs:
  request_review:
    name: "#${{ github.event.pull_request.number }}"
    runs-on: ubuntu-latest
    steps:
      - name: Request review from team if not already requested
        env:
          # To request a team review, the token must have the `read:org` scope &
          # "Members" organisation permissions (read). We can only grant repo scoped
          # permissions to the default GITHUB_TOKEN, so have to use a fine-grained personal access
          # token (PAT) to gain the required permissions.
          # See: - https://docs.github.com/en/rest/pulls/review-requests?apiVersion=2026-03-10#request-reviewers-for-a-pull-request
          #      - https://docs.github.com/en/rest/teams/teams?apiVersion=2026-03-10#list-teams
          GH_TOKEN: ${{ secrets.REVIEW_REQUESTER_WORKFLOW_TOKEN}}
        run: |
          TEAM="dev-team"
          GH_API_ENDPOINT="repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/requested_reviewers"
          
          if gh api "$GH_API_ENDPOINT" --jq '.teams[].slug' | grep -qx "$TEAM"; then
            echo "Review already requested from ${TEAM}; nothing to do."
          else
            # Only print the ordinary command output if it fails. Under normal circumstances, keep the output quiet.
            # Always allow stderr to go straight to the console, so that any error messages are visible.
            OUTPUT=$(gh api "$GH_API_ENDPOINT" -f "team_reviewers[]=$TEAM")  
            if [ $? -ne 0 ]; then
              echo "$OUTPUT"
              exit 1  # non-zero exit code to indicate workflow failure
            fi
            echo "\nRequested review from ${TEAM}."
          fi
```
We run this within an organisation, and you need read permission to the orgs members & teams to work, hence why it doesn't use the default `GITHUB_TOKEN`
